### ISSUE TITLE
Defining a derived attribute "name" and a method "setName" causes the containing class to become abstract when generating Java code.

### ISSUE THREAD

**Issue by Nava2** [umple/umple#428]
Originally reported on Google Code with ID 530
Owned by `tylerjdmcconnell`

---

**What steps will reproduce the problem?**
1. Paste the following code in Umple Online and generate Java code for it.

class Person {
  String firstName = "Tyler";
  String name = { "Your first name is "+getFirstName() };

  public void setName(String newName)
  {
    setFirstName(newName);
  }  
}

What is the expected output?

The expected output is a concrete Java class with the methods getFirstName(), setFirstName(String
aFirstName), getName(), and setName(String newName) - along with the other auto generated
methods. Each method's implementation should be as expected based on the Umple code
below.

What is the actual output?

The actual output is an abstract Java class with the methods getFirstName(), setFirstName(String
aFirstName), and getName() - along with the other auto generated methods. That is,
the Java class has been made abstract even though it was never defined as such, and
the setName(String newName) method is not created in the generated class.

Other notes:
- This problem only occurs for Java code generation

---

Reported by `tylerjdmcconnell` on 2014-03-14 15:27:33

**Comment by Nava2** [comment]
Reported by `tylerjdmcconnell` on 2014-03-14 15:27:55
- **Blocking**: #421

**Comment by Nava2** [comment]
Reported by @umple on 2014-03-21 18:19:51
- **Labels added**: Priority-High
- **Labels removed**: Priority-Medium

**Comment by Nava2** [comment]
Reported by `tylerjdmcconnell` on 2014-04-04 19:17:09

**Comment by Nava2** [comment]
As per my conversation with Andrew and Miguel, we believe the expected behaviour for
this bug is to:
 1) Remove any defined mutator/accessor methods which are named in conflict with a
derived attribute during compilation. For example, if a derived attribute is named
"class" and a method named "getClass" or "setClass" exists, then the compiler will
remove the "getClass" or "setClass" method
 2) Display a compiler warning to the user stating that their conflicting method(s)
was removed and explain exactly why it was removed, with the steps the developer should
take to correct the warning. The message would be something like, "Ignoring method
named '<conflicting method name>' because there exists a derived attribute named '<derived
attribute name>'. Derived attributes should not have explicitly defined getters/setters.
Please either remove the conflicting method or turn the derived attribute to a regular
attribute."

---

Reported by `tylerjdmcconnell` on 2014-04-18 17:26:38

**Comment by Nava2** [comment]
Once this issue is complete, another issue should be created to encapsulate the effort
required to change all code producing the new warning from this issue.

---

Reported by `tylerjdmcconnell` on 2014-04-24 15:39:47

**Comment by Nava2** [comment]
As far as I can tell, this issue is fixed. The compiler ignores the user-defined methods
that have naming conflicts with umple-generated methods. 

The Java generator produces a concrete class for the example above, and leaves out
the user-defined method setName.

---

Reported by @CraigBryan on 2015-02-09 15:06:50

**Comment by vahdat-ab** [comment]
Everthing is working well except we should not remove the setName(String newName) because we don't have a attribute named name in the generated code. it's a derived attribute thefeore we are allowed to have our own defined methods.

**Comment by vahdat-ab** [comment]
Please make sure this is correct for all languages and generated the proper test cases.
